src/firmware_new/boards/frankie_v1/crc.h

```diff
@@ -0,0 +1,59 @@
+/**
+ ******************************************************************************
+ * File Name          : CRC.h
+ * Description        : This file provides code for the configuration
+ *                      of the CRC instances.
+ ******************************************************************************
+ * @attention
+ *
+ * <h2><center>&copy; Copyright (c) 2020 STMicroelectronics.
+ * All rights reserved.</center></h2>
+ *
+ * This software component is licensed by ST under Ultimate Liberty license
+ * SLA0044, the "License"; You may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at:
+ *                             www.st.com/SLA0044
+ *
+ ******************************************************************************
+ */
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef __crc_H
+#define __crc_H
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
```

Comment:
@EvanMorcom - It's worth noting here that by including these headers you also transitively include `main.h`, which I think is what you were trying to avoid?

I still think this PR is good, just pointing this out.